contracts/bonding/BondingCheckpoints.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/governance/Governor.sol";
+import "@openzeppelin/contracts/governance/extensions/GovernorSettings.sol";
+import "@openzeppelin/contracts/utils/Checkpoints.sol";
+
+import "./libraries/EarningsPool.sol";
+import "./libraries/EarningsPoolLIP36.sol";
+
+import "../IController.sol";
+import "../rounds/IRoundsManager.sol";
+import "./BondingManager.sol";
+
+contract BondingCheckpoints {
+    using Checkpoints for Checkpoints.Trace224;
+
+    uint256 public constant MAX_ROUNDS_WITHOUT_CHECKPOINT = 100;
+
+    BondingManager public immutable bondingManagerAddr;
+
+    constructor(BondingManager _bondingManager) {
+        bondingManagerAddr = _bondingManager;
+    }
+
+    // We can't lookup the "checkpoint time" from the Checkpoints lib, only the
+    // current value. So instead of checkpointing the bonded amount and
+    // delegatee we snapshot the start round of each delegator change and lookup
+    // the specific values on the separate delegatorSnapshots mapping.
+    // TODO: Consider writing our own checkpoints lib version instead that
+    // stores directly the data we want inline.
+
+    struct DelegatorInfo {
+        uint256 bondedAmount;
+        address delegatee;
+    }
+
+    struct DelegatorCheckpoints {
+        Checkpoints.Trace224 startRounds;
+        mapping(uint256 => DelegatorInfo) snapshots;
+    }
+
+    mapping(address => DelegatorCheckpoints) private delegatorCheckpoints;
+
+    Checkpoints.Trace224 private totalActiveStakeCheckpoints;
```

Comment:
We actually call `checkpointTotalActiveStake()` indirectly from `intializeRound()`, which AFAIK is guaranteed to be called on every round as long as the protocol is active. So we could not even use the checkpoints lib here, but just have a mapping from round number to the total active stake and read that.

Should we just go with that simpler approach, or are there any cases where `initializeRound()` may not be called?

Reply:
Done here: https://github.com/livepeer/protocol/pull/611/commits/1cbe31d866e5bc8c6ba29cb97f05446093b33ffa